base/internal/message.hpp

```diff
@@ -132,7 +132,11 @@ std::string inline DebugPrint(std::nullopt_t const & p)
   return "nullopt";
 }
 
-template <typename T, size_t N> inline std::string DebugPrint(T (&arr) [N])
+// Avoid calling it for string literals.
+template <typename T, size_t N,
+         typename = std::enable_if_t<!std::is_same<typename std::remove_cv<T>::type, char>::value &&
+                                     !std::is_same<typename std::remove_cv<T>::type, char16_t>::value>>
```

Comment:
Would it be better if our implementation overloads and prints these characters, by converting them into the utf-8 and using std::string?

Reply:
And it already does via other existing overloads. What happens with char16_t in C++20 will be decided after C++20.
I don't see that we need it now.

Reply:
Is it still needed with https://github.com/organicmaps/organicmaps/pull/7324 merged? I would prefer to get build errors when someone uses it inappropriately.

Reply:
Can you please explain the intention here?

Reply:
Print char[] like string "foo", not like sequence of chars "[3: f, o, o ]"

Reply:
It may be useful to know that an array is printed. Maybe ["foo"] or "foo"[] ?